### PR TITLE
i#1834 Misc fixes: Fixes tests on ARM and AArch64

### PR DESCRIPTION
**Pull request by toshipiazza**
Commit 72be5de introduced a test failure in AArch64, and a crash on Arm
(specifically Debian Stretch). Both are fixed.

### CODE REVIEW

File: suite/tests/client-interface/drx_buf-test.c [hunk 1]
```diff
@@ -178,6 +184,7 @@ GLOBAL_LABEL(FUNCNAME:)
         pop      REG_XBX
         ret
 #elif defined(AARCHXX)
+       push     {TEST_REG_ASM}
```
Comment (derekbruening) [inline suite/tests/client-interface/drx_buf-test.c]: Looks one column off

File: suite/tests/client-interface/drx_buf-test.c [hunk 2]
```diff
@@ -195,6 +202,7 @@ GLOBAL_LABEL(FUNCNAME:)
         MOV16    TEST_REG_ASM, DRX_BUF_TEST_3_ASM
         b        epilog1
     epilog1:
+       pop      {TEST_REG_ASM}
```
Comment (derekbruening) [inline suite/tests/client-interface/drx_buf-test.c]: Ditto, and the two below

**Comment by toshipiazza** [comment]
`code_api|client.timer` failed on Mac -- should I analyze? I don't think the failure is related to my changes, is this a known failure?

File: suite/tests/client-interface/drx_buf-test.c (hunk 1 above)
Comment (toshipiazza) [inline suite/tests/client-interface/drx_buf-test.c]: Done.

File: suite/tests/client-interface/drx_buf-test.c (hunk 2 above)
Comment (toshipiazza) [inline suite/tests/client-interface/drx_buf-test.c]: Done.

**Comment by derekbruening** [comment]
> code_api|client.timer failed on Mac -- should I analyze? I don't think the failure is related to my changes, is this a known failure?

The answer is always "is it in the issue tracker".  It sounds familiar, one of the non-det failures unfortunately, looks like #2003

**Comment by derekbruening** [comment]
In general, look at issues labeled Hotlist-Travis

**Comment by toshipiazza** [comment]
Hmm, I was unable to squash and merge until I merged from master. I wonder how exactly this'll look once it gets merged? Will there be a non-linear history?

**Comment by derekbruening** [comment]
> Hmm, I was unable to squash and merge until I merged from master. I wonder how exactly this'll look once it gets merged? Will there be a non-linear history?
>
> Edit: it looks like everything merged cleanly, so likely everything will work as expected.

See https://github.com/DynamoRIO/dynamorio/wiki/Workflow which talks about this and how we want merge commits in feature branches and it says "Don't worry about merge commits in feature branches: they'll be removed in the final squash-and-fast-forward-merge step onto master"

**Comment by toshipiazza** [comment]
Understood, should I close this pull request and re-open under a feature branch? Or should I use a feature branch from now on?

**Comment by derekbruening** [comment]
> Understood, should I close this pull request and re-open under a feature branch? Or should I use a feature branch from now on?

No, the private repo branch here is close enough, the feature branch will just be deleted after merging anyway (unless something is unfinished and then we'd deliberately leave it there for someone else to pick up).

> Also apologies, I didn't see your comment in my other pull request before opening this one (you mentioned there that we were moving to a feature-branch workflow)

No problem.  Note that I also sent email to dynamorio-devs announcing it and talking about "git review" changing if you re-run devsetup -- maybe lost in a filter for automated emails.  Appveyor is noisier than it should be, not behaving like the docs say it should.

**Pull request merged**